t04_refactored.py

```diff
@@ -213,9 +213,43 @@ def team_16_adv():
 ###################################################################################
 
 def team_17_adv():
-    pass
-    # TODO Add your code here
+    print("You look at the figure and it turns out to be your best friend, Katie!")
+    print("Katie wants to hand you a demonic sword.")
+    sword = input("Do you accept the sword: Yes/No: ").lower()
+    if sword == "yes":
+        print("You unlock a special power: Super Strength!")
+    else:
+        print("Katie stabs you!")
+        isDead = True
+        kill_if_dead(isDead)
+
+    print("You enter into a hallway and meet a dog.")
+    dog = input("Would you like a companion? Yes/No ")
```

Comment:
You used `.lower()` to handle case sensitivity in the earlier input line; do it again here.